zilliqa/src/scilla.rs

```diff
@@ -275,7 +275,7 @@ impl Scilla {
             "-gaslimit".to_owned(),
             gas_limit.to_string(),
             "-balance".to_owned(),
-            value.to_string(),
+            contract_balance.to_string(),
```

Comment:
Woops, my mistake for mixing this up. Can we double check the OCaml source to confirm for certain that it is expecting the txn sender's balance here?

Reply:
The related test passes after this change, but surely I'll double check.

Reply:
@JamesHinshelwood It expects to receive the calling contract's balance here. Here's the zq1 code:

https://github.com/Zilliqa/Zilliqa/blob/f0bcbb017b18e5f16f82321374b11cfc25504882/src/libData/AccountStore/AccountStoreSC.cpp#L746

Reply:
and here:
https://github.com/Zilliqa/Zilliqa/blob/f0bcbb017b18e5f16f82321374b11cfc25504882/src/libData/AccountStore/AccountStoreSC.cpp#L1620

The txn sender's balance is included in the `message`:

```rust
                let next_message = json!({
                    "_tag": message.tag,
                    "_amount": message.amount.to_string(),
                    "_sender": format!("{to_addr:#x}"),
                    "_origin": format!("{from_addr:#x}"),
                    "params": message.params,
                });
```

Reply:
and it's sent to `run` as `-imessage`:

```rust
        let args = vec![
            "-init".to_owned(),
            serde_json::to_string(&init)?,
            "-ipcaddress".to_owned(),
            self.state_server_addr(),
            "-imessage".to_owned(),
            serde_json::to_string(&msg)?,
            "-i".to_owned(),
            code.to_owned(),
            "-gaslimit".to_owned(),
            gas_limit.to_string(),
            "-balance".to_owned(),
            contract_balance.to_string(),
            "-libdir".to_owned(),
            self.lib_dir.clone(),
            "-jsonerrors".to_owned(),
            "-pplit".to_owned(),
            "true".to_owned(),
        ];
 ```